cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -124,9 +126,48 @@
    * Create a schedule which is triggered whenever at least a certain number of new partitions
    * are added to a certain dataset.
    *
-   * @param datasetName the name of the dataset in the same namespace of the app
+   * @param datasetName   the name of the dataset in the same namespace of the app
    * @param numPartitions the minimum number of new partitions added to the dataset to trigger the schedule
    * @return this {@link ScheduleBuilder}
    */
   ScheduleCreationSpec triggerOnPartitions(String datasetName, int numPartitions);
+
+  /**
+   * Create a schedule which is triggered based upon the specified program in a namespace, application, application
```

Comment:
better to say `... which is triggered when the given program in the given namespace, application and application version transitions to the given program status`